pystac/stac_object.py

```diff
@@ -86,7 +86,7 @@ def get_single_link(
         rel: Union[str, pystac.RelType],
```

Comment:
```suggestion
        rel: Optional[Union[str, pystac.RelType]],
```

We check for `rel is None` below, so we should declare that it could be optional.

Reply:
Other than the doc string edit, I didn't monkey with the `get_single_link`. But maybe we should.

Reply:
Whoops, that's my fault -- I didn't track that the docstring was for a different function. Disregard my comment.

Reply:
It seems appropriate to modify this function, though, to make it consistent with `get_links`. I'm going to implement.